src/v/cluster/rm_stm.cc

```diff
@@ -1437,6 +1437,12 @@ ss::future<bool> rm_stm::sync(model::timeout_clock::duration timeout) {
     auto ready = co_await persisted_stm::sync(timeout);
     if (ready) {
         if (_mem_state.term != _insync_term) {
+            // There is a term change, issue a checkpoint purge to followers
+            // All other replicate ops need to wait until this finishes, this is
+            // so that the followers clean up the state before we start applying
+            // changes to the in memory state.
+            co_await _wait_for_checkpoint_purge.with(
```

Comment:
we should write tx_checkpoint_purge only after tx_checkpoint if we write it on every election it increases disk pressure and makes it harder to debug because we lose the one tx_checkpoint - one tx_checkpoint_purge invariant.

by the way let's rename tx_checkpoint and tx_checkpoint_purge to send_tx_leadership and receive_tx_leadership to highlight meaning and the relation between them